test/object.test.ts

```diff
@@ -50,4 +51,79 @@ export default {
       };"
     `);
   });
+
+  it("recursively create objects", () => {
+    const mod = parseModule(
+      `
+export default {
+  foo: {
+  }
+}
+    `.trim()
+    );
+
+    // Update existing object keys
+    expect(mod.exports.default.foo).toEqual({});
+    mod.exports.default.foo.value = 1;
+    expect(mod.exports.default.foo.value).toEqual(1);
+
+    // Create nested object
+    mod.exports.default.bar = {};
+    mod.exports.default.bar.testValue = {};
+    mod.exports.default.bar.testValue.value = "a";
+
+    expect(generate(mod)).toMatchInlineSnapshot(`
+      "export default {
+        foo: {
+          value: 1,
+        },
+
+        bar: {
+          testValue: {
+            value: \\"a\\",
+          },
+        },
+      };"
+    `);
+  });
+
+  it("recursively merge objects", () => {
+    const mod = parseModule(
+      `
+export default {
+  foo: {
+  }
+}
+    `.trim()
+    );
+
+    const obj = {
+      foo: {
+        value: 1,
+      },
+
+      bar: {
+        testValue: {
+          value: "a",
+        },
+      },
+    };
+
+    // Recursively merge existing object with `obj`
+    deepMergeObject(mod.exports.default, obj);
```

Comment:
The test you provided was `deepMergeObject(mod, obj)` - which isn't the right target to merge. Does the same bug happen to your codebase?

Meanwhile it does help to identify a bug in the deep merge function tho. Thanks!

Reply:
I wasn't sure it was the way to use it, I landed on the same `Cannot set property 'value' on type undefined` error, so I thought this was an equivalent.

By using your modified version of `deepMergeObject` in my codebase, I get the expected result!

Thank you so much for digging in! 🙏